src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -450,6 +450,12 @@ private BoundForEachStatement BindForEachPartsWorker(BindingDiagnosticBag diagno
 
             CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
             Conversion elementConversionClassification = this.Conversions.ClassifyConversionFromType(inferredType.Type, iterationVariableType.Type, isChecked: CheckOverflowAtRuntime, ref useSiteInfo, forCast: true);
+            
+            if (elementConversionClassification.Kind != ConversionKind.Identity && IterationVariable.RefKind == RefKind.Ref && builder.CurrentPropertyGetter.RefKind == RefKind.Ref)
```

Comment:
This might not handle `foreach (ref readonly var x in ...)` properly. Where do we do the check to report bad conversions on `ref var x = ref ...`?

Reply:
This is inspired by Binder_Statements.cs line 1965 if that is what you are asking?

Reply:
I took a closer look it does indeed not show the error in this case, I will test for RefReadOnly too to resolve it